zml/platform.zig

```diff
@@ -94,7 +94,40 @@ pub const Platform = struct {
     /// Returns the Profiler for this API.
     /// Not all platform have a profiling api, for those the profiler object will do nothing.
     /// Platforms with known profiler extensions: cuda, xpu
-    pub fn getProfiler(self: Platform, options: pjrt_core.Profiler.Options) pjrt_core.Profiler {
+    pub fn getProfiler(self: Platform, options: pjrt.Profiler.Options) pjrt.Profiler {
         return self.pjrt_client.getProfiler(self.pjrt_api, options);
     }
+
+    /// Suspend the current co-routine while awaiting for a pjrt event to be over.
+    pub fn awaitEvent(self: Platform, event: *pjrt.Event) !void {
+        defer event.deinit(self.pjrt_api);
+        // If we aren't in a coroutine just use the normal blocking api.
+        if (!asynk.inCoro()) {
+            return try event.await_(self.pjrt_api);
+        }
```

Comment:
I get the motivation, but it feels like implementation details leaking.

I don't really have another suggestion currently though

Reply:
I could remove it, but I find it simpler to use the blocking api if asynk is not enabled.